### PR TITLE
Fix redir expansion bug.

### LINKED ISSUE #47643: abcde: Expansion of $REDIR quoted on MacOSX breaks cp
Hit the following issue when ripping a CD:
```
usage: cp [-R [-H | -L | -P]] [-fi | -n] [-apvX] source_file target_file
       cp [-R [-H | -L | -P]] [-fi | -n] [-apvX] source_file ... target_directory
[ERROR] abcde: The following commands failed to run:
readtrack-1: cp -f returned code 64
```
Upstream already identified & fixed the issue http://abcde.einval.com/bugzilla/show_bug.cgi?id=22
Until upstream releases a new version, using the supplied patch http://abcde.einval.com/bugzilla/attachment.cgi?id=12 solves the issue, I opened a PR ( #47644 ) for it in case this is acceptable.

Sadly, building abcde with --HEAD does not work for me:
```
XXPM00355799A:~ d061778$ brew install --HEAD abcde
==> Cloning http://git.einval.com/git/abcde.git
Updating /Library/Caches/Homebrew/abcde--git
==> Checking out branch master
==> make install prefix=/usr/local/Cellar/abcde/HEAD etcdir=/usr/local/etc
Last 15 lines from /Users/d061778/Library/Logs/Homebrew/abcde/01.make:
2016-01-03 17:07:50 +0100

make
install
prefix=/usr/local/Cellar/abcde/HEAD
etcdir=/usr/local/etc

/usr/bin/install -c -d -m 755 /usr/local/Cellar/abcde/HEAD/bin
/usr/bin/install -c -m 755 abcde cddb-tool abcde-musicbrainz-tool /usr/local/Cellar/abcde/HEAD/bin
/usr/bin/install -c -d -m 755 /etc
install: chmod 755 /etc: Operation not permitted
/usr/bin/install -c -m 644 abcde.conf /etc
install: /etc/abcde.conf: Permission denied
make: *** [install] Error 71

READ THIS: https://git.io/brew-troubleshooting

These open issues may also help:
abcde: internal error: cddb-choice not recorded. https://github.com/Homebrew/homebrew/issues/46267
```

### PR DESCRIPTION
**Pull request by benjaminfrank**
See issue https://github.com/Homebrew/homebrew/issues/47643 or upstream bug report http://abcde.einval.com/bugzilla/show_bug.cgi?id=22

Closes #47643.

### CODE REVIEW

File: Library/Formula/abcde.rb [hunk 1]
```diff
@@ -22,6 +23,11 @@ class Abcde < Formula
   depends_on "vorbis-tools" => :optional
   depends_on "glyr" => :optional
 
+  patch do
+    url "http://abcde.einval.com/bugzilla/attachment.cgi?id=12"
+    sha256 "4620dd5ef7ab32b6511782da85831661ccee292fadff1f58acc3b4992486af62"
+  end
```
Comment (mikemcquaid) [inline Library/Formula/abcde.rb]: Can you please submit this patch to the upstream developers of this project (if it hasn't been already) and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!

Comment (benjaminfrank) [inline Library/Formula/abcde.rb]: This patch comes from upstream and is already included there. However, no new release yet, thus the "downporting". Using --HEAD didn't work for me either, see issue #47643.
I'll add the comment.

File: Library/Formula/abcde.rb [hunk 2]
```diff
@@ -22,6 +23,16 @@ class Abcde < Formula
   depends_on "vorbis-tools" => :optional
   depends_on "glyr" => :optional
 
+  # Fix "Expansion of $REDIR quoted on MacOSX breaks cp" bug
+  # (http://abcde.einval.com/bugzilla/show_bug.cgi?id=22)
+  # Obsolete once a new version is released as upstream already merged this
+  # patch in c024365a846faae390f23c86f32235d6209e6edf
+  # (http://git.einval.com/cgi-bin/gitweb.cgi?p=abcde.git;a=commit;h=c024365a846faae390f23c86f32235d6209e6edf)
+  patch do
+    url "http://abcde.einval.com/bugzilla/attachment.cgi?id=12"
```
Comment (mikemcquaid) [inline Library/Formula/abcde.rb]: Does http://git.einval.com/cgi-bin/gitweb.cgi?p=abcde.git;a=commitdiff_plain;h=c024365a846faae390f23c86f32235d6209e6edf apply?

Comment (benjaminfrank) [inline Library/Formula/abcde.rb]: Nope, but then it also patches the changelog file which is probably the culprit here.

Comment (mikemcquaid) [inline Library/Formula/abcde.rb]: np, just wanted to check. :+1: from me.

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew/commit/fc491657a6f33c3216e3c67531f5b2682deab3eb. Thank you for another contribution to Homebrew @benjaminfrank! :smiley_cat:

**Pull request closed without merging**